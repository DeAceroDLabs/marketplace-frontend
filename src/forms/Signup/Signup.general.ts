import { Form } from "./../form.types";
export const generalSignupForm: Form = {
  formTitle: "Información General",
  formDescription: "Comienza tu registro añadiendo tus datos generales",
  fields: [
    {
      name: "name",
      label: "Nombre",
      required: true,
      placeholder: "Nombre",
      type: "text",
      value: "",
    },
    {
      name: "phone",
      label: "Teléfono",
      required: true,
      placeholder: "+52 (000)-1111-222",
      type: "tel",
      value: "",
    },
    {
      name: "email",
      label: "Correo electrónico",
      required: true,
      placeholder: "ejemplo@deacero.com",
      type: "email",
      value: "",
      validateDomain: true,
    },
    {
      name: "password",
      label: "Contraseña",
      required: true,
      placeholder: "Contraseña",
      type: "password",
      value: "",
    },
    {
      name: "password-confirm",
      label: "Confirmar contraseña",
      required: true,
      placeholder: "Contraseña",
      type: "password",
      value: "",
    },
  ],
};
