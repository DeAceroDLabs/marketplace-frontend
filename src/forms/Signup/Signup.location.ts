import { DynamicDataContext } from "config/dynamicDataContext";
import { Form } from "./../form.types";
export const locationSignupForm: Form = {
  formTitle: "Ubicación",
  formDescription:
    "Registra la ubicación en donde seran entregados tus productos",
  fields: [
    {
      name: "zipCode",
      label: "Código Postal",
      required: true,
      placeholder: "12345",
      type: "number",
    },
    {
      name: "state",
      label: "Estado",
      required: true,
      placeholder: "Estado",
      type: "text",
      disabled: true,
    },
    {
      name: "city",
      label: "Municipio",
      required: true,
      placeholder: "Municipio",
      type: "text",
      disabled: true,
    },
    {
      name: "suburb",
      label: "Colonia",
      required: true,
      placeholder: "Colonia",
      type: "dropdown",
      dynamicOptionsProp: "dynamicData",
      optionsContext: DynamicDataContext,
    },
    {
      name: "street",
      label: "Calle",
      required: true,
      placeholder: "Calle",
      type: "text",
      size: "medium",
    },
    {
      name: "externalNumber",
      label: "Número Exterior",
      required: true,
      placeholder: "25-A",
      type: "text",
      size: "small",
    },
  ],
};
