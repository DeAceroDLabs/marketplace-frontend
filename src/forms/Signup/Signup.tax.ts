import { Form } from "./../form.types";
export const taxSignupForm: Form = {
  formTitle: "Información Fiscal",
  formDescription: "Registra los datos fiscales para facturación",
  fields: [
    {
      name: "taxIdentity",
      label: "Compañía / Nombre Legal",
      required: true,
      placeholder: "Nombre de empresa o identidad legal",
      type: "text",
      value: "",
    },
    {
      name: "zipCode",
      label: "Código Postal",
      required: true,
      placeholder: "000000",
      type: "number",
      value: "",
    },
    {
      name: "regime",
      label: "Regimen Tributario",
      required: true,
      placeholder: "ejemplo@deacero.com",
      type: "text",
      value: "Sueldos y Salarios e Ingresos Asimilados a salarios",
      disable: true,
    },
    {
      name: "state",
      label: "Estado",
      required: true,
      placeholder: "",
      type: "dropdown",
      value: "",
    },
    {
      name: "rfc",
      label: "RFC",
      required: true,
      placeholder: "AAAA000000BBB0C0",
      type: "text",
      value: "",
    },
  ],
};
