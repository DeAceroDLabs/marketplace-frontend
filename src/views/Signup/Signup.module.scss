.container {
    display: flex;
    flex-direction: row;
}

.form-container {
    width: 75%;
    margin-left: 20px;
}

.popup {
    text-align: center;
    display: flex;
    flex-direction: column;
    align-items: center;
    .popup-text {
        margin-top: 56px;
        margin-bottom: 20px;
    }
}