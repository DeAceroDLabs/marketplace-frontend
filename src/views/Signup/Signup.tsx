import { useMemo, useState } from "react";
import { FieldValues } from "react-hook-form";
import { MultiFormProvider } from "config/multiFormContext";
import { generalSignupForm } from "forms/Signup/Signup.general";
import { locationSignupForm } from "forms/Signup/Signup.location";
import { taxSignupForm } from "forms/Signup/Signup.tax";
import { Form } from "forms/form.types";
import BgSection from "components/common/BgSection";
import Section from "components/common/Section";
import View from "components/common/View";
import MultipleForm from "components/Form/MultipleForms";
import Stepper from "components/Form/Stepper";
import styles from "./Signup.module.scss";

const Signup = () => {
  const forms = [generalSignupForm, locationSignupForm, taxSignupForm];
  const [activeForm, setActiveForm] = useState(forms[0] as Form);
  const signupForms = useMemo(
    () => ({ forms, activeForm, setActiveForm }),
    // eslint-disable-next-line
    [activeForm]
  );

  const onSubmit = (data: FieldValues) => {
    console.log(data);
  };

  return (
    <MultiFormProvider value={signupForms}>
      <View>
        <div className={styles.container}>
          <Section>
            <div className={styles["form-container"]}>
              <MultipleForm inputForms={forms} onSubmit={onSubmit} />
            </div>
          </Section>
          <BgSection color="primary" orientation="vertical" position="right">
            <Stepper
              title="Es muy fácil y rápido"
              steps={forms.map((form) => form.formTitle)}
            />
          </BgSection>
        </div>
      </View>
    </MultiFormProvider>
  );
};

export default Signup;
