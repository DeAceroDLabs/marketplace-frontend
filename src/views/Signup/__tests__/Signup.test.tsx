import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Signup from "../Signup";

describe("Signup", () => {
  it("loads login without exploding", () => {
    const view = render(
      <BrowserRouter>
        <Signup />
      </BrowserRouter>
    );
    expect(view).toMatchSnapshot();
  });
});
