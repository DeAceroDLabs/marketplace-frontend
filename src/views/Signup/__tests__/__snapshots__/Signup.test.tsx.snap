// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup loads login without exploding 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div />
      </div>
      <div
        class="body"
      >
        <div>
          <div
            class="container"
          >
            <div
              class="container"
            >
              <div
                class="title margin-title"
              >
                 
              </div>
              <div
                class="body"
              >
                 
                <div
                  class="form-container"
                >
                  <div
                    class="container"
                  >
                    <form
                      class="form-container"
                    >
                      <div
                        class="container"
                      >
                        <div
                          class="title margin-title"
                        >
                          Información General
                           
                        </div>
                        <div
                          class="body"
                        >
                           
                          <p
                            class="form-description"
                          >
                            Comienza tu registro añadiendo tus datos generales
                          </p>
                          <div
                            class="form-container"
                          >
                            <div
                              class="field-container"
                            >
                              <div
                                class="container"
                              >
                                <label>
                                  Nombre
                                </label>
                                <input
                                  class="input"
                                  name="name"
                                  placeholder="Nombre"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                            <div
                              class="field-container"
                            >
                              <div
                                class="container"
                              >
                                <label>
                                  Teléfono
                                </label>
                                <input
                                  class="input"
                                  name="phone"
                                  placeholder="+52 (000)-1111-222"
                                  type="tel"
                                  value=""
                                />
                              </div>
                            </div>
                            <div
                              class="field-container"
                            >
                              <div
                                class="container"
                              >
                                <label>
                                  Correo electrónico
                                </label>
                                <input
                                  class="input"
                                  name="email"
                                  placeholder="ejemplo@deacero.com"
                                  type="email"
                                  value=""
                                />
                              </div>
                            </div>
                            <div
                              class="field-container"
                            >
                              <div
                                class="container"
                              >
                                <label>
                                  Contraseña
                                </label>
                                <input
                                  class="input"
                                  name="password"
                                  placeholder="Contraseña"
                                  type="password"
                                  value=""
                                />
                              </div>
                            </div>
                            <div
                              class="field-container"
                            >
                              <div
                                class="container"
                              >
                                <label>
                                  Confirmar contraseña
                                </label>
                                <input
                                  class="input"
                                  name="password-confirm"
                                  placeholder="Contraseña"
                                  type="password"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                           
                        </div>
                      </div>
                    </form>
                    <div
                      class="button-container"
                    >
                      <button
                        class="button primary activeClass"
                      >
                        siguiente
                      </button>
                    </div>
                  </div>
                </div>
                 
              </div>
            </div>
            <div
              class="vertical"
            >
              <div
                class="content primary right"
              >
                <div
                  class="stepper-wrapper"
                >
                  <div
                    class="stepper"
                  >
                    <div
                      class="title"
                    >
                      Es muy fácil y rápido
                    </div>
                    <div
                      class="stepper-selector"
                    >
                      <div
                        class="line"
                      />
                      <div
                        class="step"
                      >
                        <div
                          class="step-checkbox"
                        >
                          <div
                            class="selected"
                          />
                        </div>
                        <div
                          class="step-title"
                        >
                          Información General
                        </div>
                      </div>
                      <div
                        class="step"
                      >
                        <div
                          class="step-checkbox"
                        />
                        <div
                          class="step-title"
                        >
                          Ubicación
                        </div>
                      </div>
                      <div
                        class="step"
                      >
                        <div
                          class="step-checkbox"
                        />
                        <div
                          class="step-title"
                        >
                          Información Fiscal
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div />
    </div>
    <div
      class="body"
    >
      <div>
        <div
          class="container"
        >
          <div
            class="container"
          >
            <div
              class="title margin-title"
            >
               
            </div>
            <div
              class="body"
            >
               
              <div
                class="form-container"
              >
                <div
                  class="container"
                >
                  <form
                    class="form-container"
                  >
                    <div
                      class="container"
                    >
                      <div
                        class="title margin-title"
                      >
                        Información General
                         
                      </div>
                      <div
                        class="body"
                      >
                         
                        <p
                          class="form-description"
                        >
                          Comienza tu registro añadiendo tus datos generales
                        </p>
                        <div
                          class="form-container"
                        >
                          <div
                            class="field-container"
                          >
                            <div
                              class="container"
                            >
                              <label>
                                Nombre
                              </label>
                              <input
                                class="input"
                                name="name"
                                placeholder="Nombre"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="field-container"
                          >
                            <div
                              class="container"
                            >
                              <label>
                                Teléfono
                              </label>
                              <input
                                class="input"
                                name="phone"
                                placeholder="+52 (000)-1111-222"
                                type="tel"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="field-container"
                          >
                            <div
                              class="container"
                            >
                              <label>
                                Correo electrónico
                              </label>
                              <input
                                class="input"
                                name="email"
                                placeholder="ejemplo@deacero.com"
                                type="email"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="field-container"
                          >
                            <div
                              class="container"
                            >
                              <label>
                                Contraseña
                              </label>
                              <input
                                class="input"
                                name="password"
                                placeholder="Contraseña"
                                type="password"
                                value=""
                              />
                            </div>
                          </div>
                          <div
                            class="field-container"
                          >
                            <div
                              class="container"
                            >
                              <label>
                                Confirmar contraseña
                              </label>
                              <input
                                class="input"
                                name="password-confirm"
                                placeholder="Contraseña"
                                type="password"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                         
                      </div>
                    </div>
                  </form>
                  <div
                    class="button-container"
                  >
                    <button
                      class="button primary activeClass"
                    >
                      siguiente
                    </button>
                  </div>
                </div>
              </div>
               
            </div>
          </div>
          <div
            class="vertical"
          >
            <div
              class="content primary right"
            >
              <div
                class="stepper-wrapper"
              >
                <div
                  class="stepper"
                >
                  <div
                    class="title"
                  >
                    Es muy fácil y rápido
                  </div>
                  <div
                    class="stepper-selector"
                  >
                    <div
                      class="line"
                    />
                    <div
                      class="step"
                    >
                      <div
                        class="step-checkbox"
                      >
                        <div
                          class="selected"
                        />
                      </div>
                      <div
                        class="step-title"
                      >
                        Información General
                      </div>
                    </div>
                    <div
                      class="step"
                    >
                      <div
                        class="step-checkbox"
                      />
                      <div
                        class="step-title"
                      >
                        Ubicación
                      </div>
                    </div>
                    <div
                      class="step"
                    >
                      <div
                        class="step-checkbox"
                      />
                      <div
                        class="step-title"
                      >
                        Información Fiscal
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
